src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java

```diff
@@ -163,6 +163,16 @@ public String getInlineN1qlQuery() {
 		return StringUtils.hasText(query) ? query : null;
 	}
 
+	/**
+	 * indicates if the method begins with "count"
+	 *
+	 * @return true if the method begins with "count", indicating that .count() should be called instead of one() or
+	 *         all().
+	 */
+	public boolean isCountQuery() {
+		return getName().startsWith("count");
```

Comment:
could this be a case insensitive check as well to minimize the risk of not catching it?

Reply:
Ok, good idea.